src/middlewares/error_handlers.cr

```diff
@@ -1,16 +1,74 @@
+require "http"
+require "json"
+
+struct ErrorData
+  include JSON::Serializable
+  property type : String
+  property message : String?
+  property status_code : HTTP::Status?
+
+  def initialize(@type, @message, @status_code = nil)
```

Comment:
```suggestion
  def initialize(@type : String, @message : String?, @status_code : HTTP::Status? = nil)
```

Reply:
What is the point of this? A compiler error will still result if one calls `ErrorMessage.new` with a `Int32` as the first value. In fact, I had to make some changes to the types of this as I was developing it -- I originally had it like

```crystal
  property type : String
  property message : String
  property status_code : Int32?
```
and the compiler told me I couldn't call `ErrorMessage.new(String, String?, HTTP::Status?)`, as one would expect.

Reply:
They will appear in the API docs. Maybe the compiler error is better, too.